user-api/src/main/java/com/example/auction/user/api/User.java

```diff
@@ -12,23 +12,13 @@
 
     private final UUID id;
     private final String name;
+    private final String email;
 
     @JsonCreator
-    private User(@JsonProperty("id") Optional<UUID> id, @JsonProperty("name") String name) {
-        this.id = id.orElse(null);
-        this.name = name;
-    }
-
-    public User(UUID id, String name) {
+    public User(@JsonProperty("id") UUID id, @JsonProperty("name") String name, @JsonProperty("email") String email) {
```

Comment:
Did you find it necessary to add the `@JsonProperty` annotations, or are you doing it just to be explicit?

We have enabled the Jackson parameter names extension, which can detect the names of constructor parameters compiled into the bytecode, so it shouldn't be necessary to add explicit annotations. If that isn't working, it might be that something is misconfigured.